### PR TITLE
[feature](Nereids) found rows support

### PR DESCRIPTION
**Pull request by xzj7019**
## Proposed changes

found rows support，controlled by user property:
SET PROPERTY FOR '$USER' 'enable_found_rows' = 'true';
e.g,
SET PROPERTY FOR 'root' 'enable_found_rows' = 'true';

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.List;
+import java.util.Set;
+
+/**
+ * FoundRows support.
+ */
+public class FoundRows extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalResultSink(logicalProject(logicalAggregate(logicalSubQueryAlias())))
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java:49]: why this pattern?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Analyzer.java [hunk 2]
```diff
@@ -125,6 +127,8 @@ private static List<RewriteJob> buildAnalyzeJobs(Optional<CustomTableResolver> c
             topDown(new EliminateGroupByConstant()),
             topDown(new NormalizeAggregate()),
             bottomUp(new JoinCommute()),
+            topDown(new CalcFoundRows()),
+            topDown(new FoundRows()),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Analyzer.java:131]: we call setFoundRowsPlan in CalcFoundRows rule. and getFoundRowsPlan in FoundRows, so we need apply FoundRows in the head of CalcFoundRows

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                .thenApply(ctx -> {
+                    String user = ctx.cascadesContext.getConnectContext().getQualifiedUser();
+                    if (user == null || !Env.getCurrentEnv().getAuth().isEnableFoundRows(user)
+                            || ctx.cascadesContext.getConnectContext().getFoundRowsPlan() == null) {
+                        return null;
+                    }
+                    LogicalResultSink<LogicalProject<LogicalAggregate<LogicalSubQueryAlias<Plan>>>> rs = ctx.root;
+                    LogicalProject<LogicalAggregate<LogicalSubQueryAlias<Plan>>> project = rs.child();
+                    if (project.getProjects().size() != 1) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java:58]: we need add more comment to explain limitations

**Comment by xzj7019** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java (hunk 1 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java:49]: second sql pattern is a fixed select count(*) wrapping the former sql without added limit.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Analyzer.java (hunk 2 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Analyzer.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java (hunk 3 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FoundRows.java:59]: done

**Pull request closed without merging**